### PR TITLE
docs: first pass at sphinx-built API docs

### PR DESCRIPTION
**Pull request by chipaca**
To build these, `pip install sphinx`, and then e.g. `make -C docs html`; generated docs will be in e.g. `docs/_build/html`.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# -- Options for sphinx.ext.autodoc ------------------------------------------
+
+# This value controls how to represents typehints. The setting takes the
+# following values:
+#     'signature' – Show typehints as its signature (default)
+#     'description' – Show typehints as content of function or method
+#     'none' – Do not show typehints
+autodoc_typehints = 'description'
+
```
Comment (jameinel) [inline docs/conf.py:69]: I missed that this doesn't have the autoclass_content = 'both' that you had me add before.

That needs to be added, but the rest look good to me. I'm sure there is more tweaking we can do over time, but having something is going to be a lot better than what we have now. :)

Comment (chipaca) [reply docs/conf.py:69]: I was waiting for your confirmation that that change had DTRT for you :slightly_smiling_face:  I'm taking this as a ‘yes’, and have pushed that change in.

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from pathlib import Path
+import sys
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'The Operator Framework'
+copyright = '2020, Canonical Ltd'
```
[the remark this answers is not in the document]
Comment (chipaca) [reply docs/conf.py:17]: I thought copyright tracked the date of last change

**Comment by jameinel** [comment]
I thought copyright tracked the date of the *first* change so you could
clearly state that you had generated this before the competitors.
(Certainly I remember getting legal telling us that we didn't have to keep
updating things to be a range, but make sure that we had the start date
correct.)

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (facundobatista) [inline docs/Makefile:1]: What about having a "build_docs" script at project's root?

File: docs/conf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (facundobatista) [reply docs/conf.py:17]: Copyright for the files must "start" when file created originally, and then "end" (if different than "start") always updated when the file is changed.

But the Copyright here is "of the whole project", so it should start from the lowest year number, IMO.

Also, `Ltd` is `Ltd.`

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+
+
+# -- Options for sphinx.ext.todo ---------------------------------------------
+
+#  If this is True, todo and todolist produce output, else they
+#  produce nothing. The default is False.
+todo_include_todos = True
```
Comment (facundobatista) [inline docs/conf.py:59]: Do we want TODOs in final docs?

File: docs/Makefile (hunk 3 above)
Comment (chipaca) [reply docs/Makefile:1]: yeah, probably easier

File: docs/conf.py (hunk 4 above)
Comment (chipaca) [reply docs/conf.py:59]: ok

Comment (chipaca) [reply docs/conf.py:59]: (i mean, no we don't; also OK about the copyright one that githb doesn't let me comment on)

File: docs/conf.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 # -- Project information -----------------------------------------------------
 
 project = 'The Operator Framework'
-copyright = '2020, Canonical Ltd'
+copyright = '2019, Canonical Ltd.'
```
Comment (facundobatista) [inline docs/conf.py:17]: `2019-2020`

**Pull request merged**